### PR TITLE
Fixes in evaluation_basic.rst

### PR DESCRIPTION
**Pull request by rasbt**
## What does this PR do?

Adds documentation fixes. The same doc used `pl.LightningModule` but was missing the `pl.` for the `Trainer`.

### CODE REVIEW

File: docs/source-pytorch/common/evaluation_basic.rst [hunk 1]
```diff
@@ -124,7 +124,8 @@ To run the validation loop, pass in the validation set to **.fit**
 
    train_loader = DataLoader(train_set)
    valid_loader = DataLoader(valid_set)
+   model = LitAutoEncoder(...)
 
    # train with both splits
-   trainer = Trainer()
+   trainer = pl.Trainer()
```
Comment (awaelchli) [inline docs/source-pytorch/common/evaluation_basic.rst:130]: Hmm, we use `Trainer()` or `L.Trainer()` everywhere else in the docs. I think it doesn't make much sense to add the `pl.` if we're not importing it anywhere.

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/common/evaluation_basic.rst (hunk 1 above)
Comment (rasbt) [reply docs/source-pytorch/common/evaluation_basic.rst:130]: Yes, maybe we should explicitly import it or remove it in both places. Right now it's just inconsistent.
Screenshot 2023-10-23 at 6 55 00 PM

Comment (rasbt) [reply docs/source-pytorch/common/evaluation_basic.rst:130]: > Hmm, we use Trainer() or L.Trainer() everywhere else in the docs

In the Level 1 Beginner docs it's `pl`: https://lightning.ai/docs/pytorch/stable/model/train_model_basic.html
Screenshot 2023-10-23 at 6 56 50 PM

Comment (Borda) [reply docs/source-pytorch/common/evaluation_basic.rst:130]: lets drop it in both places :)

Comment (awaelchli) [reply docs/source-pytorch/common/evaluation_basic.rst:130]: I'll do the rest here: #18869
We can still get the small fix from @rasbt in like this IMO.

Comment (awaelchli) [reply docs/source-pytorch/common/evaluation_basic.rst:130]: 3 and 4 are the goal for our docs. 4 depends on the context, sometimes things need to be imported that aren't available from the top lightning module, so you need to go deeper.

Anything with pytorch_lightning (1 and 2) definitely not, this should not be found in any of our recent docs :)

**Review by Borda — APPROVED** [review]

**Pull request merged**